### PR TITLE
chore(react-query): drop support for React17

### PR DESCRIPTION
**Pull request by illuzan**
Closes: #4683

### CODE REVIEW

**Comment by TkDodo** [comment]
Thank you. I think there are a couple of things missing:

- look for `REACTJS_VERSION` string
- most things in `jest.setup.ts` can go:

https://github.com/TanStack/query/blob/4ae99561ca3383d6de3f4aad656a49ba4a17b57a/packages/react-query/jest.setup.ts#L9-L39

- this as well: https://github.com/TanStack/query/blob/c376158afaddf7cd92a85491c3f18c1b123efe4b/packages/react-query/src/__tests__/ssr-hydration.test.tsx#L17
- we can always use `ReactDOM.hydrate`:
https://github.com/TanStack/query/blob/c376158afaddf7cd92a85491c3f18c1b123efe4b/packages/react-query/src/__tests__/ssr-hydration.test.tsx#L19-L35
- this is also a leftover that can go:
https://github.com/TanStack/query/blob/11f08c37d56c8aa5d9b28d8422842bc1a147c487/package.json#L38

**Comment by TkDodo** [comment]
@illuzan will you continue to work on this PR or should someone else take over?

**Comment by illuzan** [comment]
Yes, I will work on this I was just busy with work stuff.

**Comment by illuzan** [comment]
@TkDodo I have pushed an update, can you take a look?

**Comment by TkDodo** [comment]
there are conflicts with v5, can you please fix them?

File: packages/react-query/src/__tests__/useQuery.test.tsx [hunk 1]
```diff
@@ -2409,7 +2409,7 @@ describe('useQuery', () => {
     await waitFor(() => rendered.getByText('count: 2'))
 
     // Should be 2 / 3 instead of 5, uSES batches differently
```
Comment (TkDodo) [inline packages/react-query/src/__tests__/useQuery.test.tsx:2411]:
```suggestion
    // Should be 3 instead of 5
```

**Review by TkDodo — APPROVED** [review]

**Comment by illuzan** [comment]
Looks like I messed up pushing it again.

**Pull request merged**